arex-storage-web-api/src/main/java/com/arextest/storage/service/DatabaseParseService.java

```diff
@@ -151,13 +164,32 @@ public static TableSchema parse(String sql) {
         } catch (Throwable e) {
             LOGGER.warn("parse sql error, sql: {}", sql, e);
         }
+        long totalTimeNanos = System.nanoTime() - startTimeNanos;
+        recordParseTimeAndLength(totalTimeNanos, sql.length());
         return tableSchema;
     }
 
-    static String getAction(Statement statement) {
+    private String getAction(Statement statement) {
         if (statement instanceof Select) {
             return "Select";
         }
         return statement.getClass().getSimpleName();
     }
+
+    private void recordParseTimeAndLength(long duration, int sqlLength) {
+        if (CollectionUtils.isEmpty(metricListenerList)) {
+            return;
+        }
+        long durationMillis = nanosToMillis(duration);
+        for (MetricListener metricListener : metricListenerList) {
+            metricListener.recordTime(SQL_PARSE_TIME_METRIC_NAME, Collections.emptyMap(), durationMillis);
```

Comment:
Hiding points requires tags, Can refer to
com.arextest.storage.metric.AgentWorkingMetricService#recordEntryTime

Reply:
Using appId as tag?